### DISCUSSION TITLE
Generate the json file correctly

### DISCUSSION THREAD

**Discussion by Antigen-1** [openai/whisper#1115] [General]
`NaN` from javascript is not a valid token in json, but is sometimes used as values in the output files.

That certainly confuses json parsers which abide by  the standard strictly.

**Comment by jongwook** [comment]
Good point. Are you seeing `NaN`s in the word probabilities?

**Reply by Antigen-1** [reply]
I'm not sure. I have only found a `NaN` in a `avg_logprob` key-value pair.

**Reply by BillyCroan** [reply]
Yes.  This is what we are seeing:
"avg_logprob": NaN,

It's, been broken like this since I started using it in 2023.  What's the official channel to report bugs.  I'd say making a .json file that isn't truely compliant with json is a problem.

if you run whisper's json through jq. it will clean it up (for those needing a short term fix) but gross.  That should not be necessary

I'd also love some documentation on the exact meaning of all fields in whisper's json output, given it is the most informative output whisper can produce.

I am using the json output in an html/javascript page to highlight the current sentence as its being played back on that same page.  and my javascript breaks when it tries to read the larger json files because more often than not they have this invalid NaN usage, which I think should be "null" or "none".   or 0 or just leaving that key off would be fine to most I think.
